plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/schema/SQLSchemaManager.java

```diff
@@ -113,19 +122,19 @@ private void upgradeSchemaVersion(DBRProgressMonitor monitor, Connection connect
         for (int curVer = currentSchemaVersion; curVer < schemaVersionActual; curVer++) {
             int updateToVer = curVer + 1;
             Reader ddlStream = scriptSource.openSchemaUpdateScript(monitor, updateToVer);
-            if (ddlStream != null) {
-                log.debug("Update schema " + schemaId + " version from " + curVer + " to " + updateToVer);
-                try {
-                    executeScript(monitor, connection, ddlStream, true);
-                } catch (Exception e) {
-                    log.warn("Error updating " + schemaId + " schema version from " + curVer + " to " + updateToVer, e);
-                    throw e;
-                } finally {
-                    ContentUtils.close(ddlStream);
-                }
+            if (ddlStream == null) {
+                continue;
+            }
```

Comment:
The `try-with-resources` will do better in this situation. It's `null`-aware, so you can safely do the following:

```java
try (Reader ddlStream = scriptSource.openSchemaUpdateScript(monitor, updateToVer)) {
    ...
}